build.cake

```diff
@@ -470,24 +535,34 @@ Task("TestPublished")
     .Does(() =>
 {
     var project = buildPlan.MainProject;
-    foreach (var framework in buildPlan.Frameworks)
+    var scriptsToTest = new string[] {"OmniSharp", "OmniSharp.Core"};
+    foreach (var script in scriptsToTest)
     {
-        // Skip testing mono executables
-        if (!IsRunningOnWindows() && !framework.Equals("netcoreapp"))
-        {
-            continue;
-        }
-        var outputFolder = $"{publishFolder}/{project}/default/{framework}";
-        var process = StartAndReturnProcess($"{outputFolder}/{project}",
-            new ProcessSettings
+        // Use System.Diagnostics.Process to access ProcessId
+        var process = System.Diagnostics.Process.Start(
+            new ProcessStartInfo($"{shell}",
+                $"{shellArgument}  {scriptFolder}/{script} -s {sourceFolder}/{project} --stdio")
             {
-                Arguments = $"-s {sourceFolder}/{project} --stdio",
+                UseShellExecute = false
             });
         // Wait 10 seconds to see if project terminates early with error
         bool exitsWithError = process.WaitForExit(10000);
         if (exitsWithError)
         {
-            throw new Exception($"Failed to run {project} / default / {framework}");
+            throw new Exception($"Failed to run {script}");
+        }
+        // Need to kill script recursively on Windows
+        if (IsRunningOnWindows())
+        {
```

Comment:
In `dotnet-projectmodel-server` we have a mechanism to let the server terminate itself. Here's how it works: the parent host's pid is sent to the server when it is started, in our case is the editor's pid. The server monitor the editor process and terminate when editor exits. Too big a change here though.

@david-driscoll how does atom deal with this?